packages/elements/list/src/getListInsertFragment.ts

```diff
@@ -1,51 +1,83 @@
 import { findNode } from '@udecode/plate-common';
 import {
   getPlatePluginOptions,
+  PlatePluginOptions,
   SPEditor,
   TDescendant,
 } from '@udecode/plate-core';
-import { Path, Transforms } from 'slate';
+import { Node, NodeEntry, Path, Transforms } from 'slate';
 import { ELEMENT_LI, ELEMENT_OL, ELEMENT_UL } from './defaults';
 
 export const getListInsertFragment = (editor: SPEditor) => {
   const { insertFragment } = editor;
 
-  return (fragment: TDescendant[]) => {
-    const li = getPlatePluginOptions(editor, ELEMENT_LI);
-    const ul = getPlatePluginOptions(editor, ELEMENT_UL);
-    const ol = getPlatePluginOptions(editor, ELEMENT_OL);
+  const li = getPlatePluginOptions(editor, ELEMENT_LI);
+  const ul = getPlatePluginOptions(editor, ELEMENT_UL);
+  const ol = getPlatePluginOptions(editor, ELEMENT_OL);
 
-    const liEntry = findNode(editor, { match: { type: li.type } });
+  const isListRoot = (node: TDescendant): boolean =>
+    [ul.type, ol.type].includes(node.type);
```

Comment:
I would rename this to `isList`. In the codebase we use `isListNested` if the list has a list ancestor, as opposite to `isListRoot` if the list has no list ancestor